File: freecodecamp/daily-code-challenge/2026-04-02.py
def capitalize_fibonacci(s):
    # Generate Fibonacci indices up to len(s)
    fib = [0, 1]
    while fib[-1] < len(s):
        fib.append(fib[-1] + fib[-2])
    
    fib_set = set(fib)

    result = []
    
    for i, ch in enumerate(s):
        if not ch.isalpha():
            result.append(ch)
        elif i in fib_set:
            result.append(ch.upper())
        else:
            result.append(ch.lower())

    return ''.join(result)

File: freecodecamp/daily-code-challenge/2026-04-02.js
function capitalizeFibonacci(str) {
  // Generate Fibonacci indices
  let fib = [0, 1];
  while (fib[fib.length - 1] < str.length) {
    fib.push(fib[fib.length - 1] + fib[fib.length - 2]);
  }

  let fibSet = new Set(fib);

  let result = "";

  for (let i = 0; i < str.length; i++) {
    let ch = str[i];

    if (!/[a-zA-Z]/.test(ch)) {
      result += ch;
    } else if (fibSet.has(i)) {
      result += ch.toUpperCase();
    } else {
      result += ch.toLowerCase();
    }
  }

  return result;
}